Assets/MRTK/Core/Utilities/XRSubsystemHelpers.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#if UNITY_2019_3_OR_NEWER
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.XR;
+#else
+using UnityEngine.Experimental.XR;
+#endif
+
+namespace Microsoft.MixedReality.Toolkit.Utilities
+{
+    /// <summary>
+    /// A helper class to provide easier access to active Unity XR SDK subsystems.
+    /// </summary>
+    /// <remarks>These properties are only valid for the XR SDK pipeline.</remarks>
+    public static class XRSubsystemHelpers
+    {
+        private static XRInputSubsystem inputSubsystem = null;
+#if UNITY_2019_3_OR_NEWER
+        private static readonly List<XRInputSubsystem> XRInputSubsystems = new List<XRInputSubsystem>();
+#endif // UNITY_2019_3_OR_NEWER
+
+        /// <summary>
+        /// The XR SDK input subsystem for the currently loaded XR plug-in.
```

Comment:
Just to make sure I understand - currently loaded == the only running one right? Would it ever be possible for multiple subsystems to be running? (i.e. if there could be some world where multiple mesh providers would be running and providing meshes?)

Reply:
According to the AR Subsystems documentation at https://docs.unity3d.com/Packages/com.unity.xr.arsubsystems@2.1/manual/index.html:

>You may only have a single subsystem per platform.

It's referring to the scenario of a specific type of subsystem, not all subsystems, but does appear to confirm that you can only have one mesh provider running at a time.